backend/src/services/compras/index.js

```diff
@@ -49,27 +57,85 @@ const updateCompra = (request, response) => {
             .send({ message: 'Missing path param id.' });
     }
 
-    // TODO: findById from DB
-    const existentCompra = listCompras().find(c => c.id == compraId)
-    if (!existentCompra) {
-        return response
-            .status(404)
-            .send({ message: `Compra ${compraId} not found.` });
-    }
-
+    const existentCompra = getCompraById(compraId, response);
```

Comment:
Não cheguei a testar, mas acho que a lógica depois desse ponto pode dar pau caso a compra não exista e role um 404. Tava lendo aqui e acho que dá pra fazer assim:

```suggestion
    const existentCompra = getCompraById(compraId, response);
    if (response.headersSent) {
      return;
    }
```

Reply:
Eu não sei se testar pelo `headersSent `é uma boa, mas eu tava pensando em talvez passar um callback pro `getCompraById `do que processar depois, ai não precisamos ficar controlando em todo o método o header

Vou dar um tapa aqui e posto a sugestão

Reply:
Bom, acabei conversando com a @tyemy e ela sugeriu de separar as responsabilidades, aí esse problema seria evitado. Criei um controller e separei o que é negócio do que é response, status, etc.

Agora pra ficar melhor seria legal criar umas exceptions hehe